linera-core/src/client.rs

```diff
@@ -1370,18 +1416,265 @@ where
         Ok(certificate)
     }
 
+    #[tracing::instrument(level = "trace", skip_all)]
+    pub async fn synchronize_chain_state(
```

Comment:
Moving this code out of `local_node.rs` to `client.rs` is not ideal.

Reply:
And if we can move it back it would be good to do it in this PR, to reduce the diff.

Reply:
It seems like that, and everything that is used by it, can also be moved to `local_node.rs`?

Reply:
Oh ok! If we're cool with that, then I agree that's better. Let me try it

Reply:
Ok, so I tried, and I think trying to move this stuff to `local_node` instead will be much worse in the amount of code we'll have to move, and also we'll need to move stuff like `known_committees`, and other validator specific things, as well as `process_certificate`, etc. So leaving things like this seems like the lesser of evils

Reply:
I guess I was trying to understand the reason to begin with. Maybe that's fine